src/types/config.ts

```diff
@@ -0,0 +1,35 @@
+import { DataSourceJsonData } from '@grafana/data';
+
+export interface CHConfig extends DataSourceJsonData {
+  username: string;
+  server: string;
+  protocol: Protocol;
+  port: number;
+  defaultDatabase?: string;
+  tlsSkipVerify?: boolean;
+  tlsAuth?: boolean;
+  tlsAuthWithCACert?: boolean;
+  secure?: boolean;
+  validate?: boolean;
+  timeout?: string;
+  queryTimeout?: string;
+  customSettings?: CHCustomSetting[];
+  enableSecureSocksProxy?: boolean;
+}
+
+export interface CHCustomSetting {
```

Comment:
This was here from v3, but I don't see anything in the parent type for this. I'm not sure if it needs to be called "custom settings" at all since the majority of these options are already unique to our plugin.